### PR TITLE
Simplify control flow in omrsysinfo_get/set_limit

### PR DESCRIPTION
**Pull request by dnakamura**
Signed-off-by: Devin Nakamura <devinn@ca.ibm.com>

### CODE REVIEW

**Comment by dnakamura** [comment]
@genie-omr build all

**Comment by 0xdaryl** [comment]
@youngar or @rwy0717 : would you mind reviewing this please?

File: port/unix/omrsysinfo.c [hunk 1]
```diff
@@ -3141,73 +3141,43 @@ omrsysinfo_get_hostname(struct OMRPortLibrary *portLibrary, char *buffer, size_t
...
+omrsysinfo_set_limit(struct OMRPortLibrary *portLibrary, uint32_t resourceID, uint64_t limit)
+{
+	int rc = OMRPORT_LIMIT_UNKNOWN;
+
+	Trc_PRT_sysinfo_set_limit_Entered(resourceID, limit);
+	Trc_PRT_sysinfo_set_limit_Exit(rc);
+	return rc;
+}
+
+#else
```
Comment (rwy0717) [inline port/unix/omrsysinfo.c:3168]:
```suggestion
#else /* defined(OMRZTPF) */
```

File: port/unix/omrsysinfo.c [hunk 2]
```diff
@@ -3252,144 +3239,105 @@ omrsysinfo_get_limit(struct OMRPortLibrary *portLibrary, uint32_t resourceID, ui
...
+#if defined(AIXPPC)
+		struct vario myvar;
+
+		myvar.v.v_fullcore.value = limit;
+		rc = sys_parm(SYSP_SET, SYSP_V_FULLCORE, &myvar);
+		if (-1 == rc) {
+			portLibrary->error_set_last_error(portLibrary, errno, findError(errno));
+			Trc_PRT_sysinfo_sysparm_error(errno);
+		}
+#else
```
Comment (rwy0717) [inline port/unix/omrsysinfo.c:3268]:
```suggestion
#else /* defined(AIXPPC) */
```

File: port/unix/omrsysinfo.c [hunk 3]
```diff
@@ -3252,144 +3239,105 @@ omrsysinfo_get_limit(struct OMRPortLibrary *portLibrary, uint32_t resourceID, ui
...
+		/* unsupported so return error */
 		rc = OMRPORT_LIMIT_UNKNOWN;
-#endif /* !defined(OMRZTPF) */
-	default:
-		break;
-	}
+#endif
```
Comment (rwy0717) [inline port/unix/omrsysinfo.c:3271]:
```suggestion
#endif /* defined(AIXPPC) */
```

File: port/unix/omrsysinfo.c [hunk 4]
```diff
@@ -3252,144 +3239,105 @@ omrsysinfo_get_limit(struct OMRPortLibrary *portLibrary, uint32_t resourceID, ui
...
+					Trc_PRT_sysinfo_setrlimit_error(resource, limit, findError(errno));
+				}
+				else {
+					limit = maxFiles;
+				}
 			}
-#else
-			/* unsupported so return error */
-			rc = OMRPORT_LIMIT_UNKNOWN;
 #endif
```
Comment (rwy0717) [inline port/unix/omrsysinfo.c:3322]:
```suggestion
#endif /* defined(OSX) */
```

File: port/unix/omrsysinfo.c [hunk 5]
```diff
@@ -3141,73 +3141,43 @@ omrsysinfo_get_hostname(struct OMRPortLibrary *portLibrary, char *buffer, size_t
 	return 0;
 }
 
+
+/* Limits are unsupported on z/TPF */
+#if defined(OMRZTPF)
```
Comment (rwy0717) [inline port/unix/omrsysinfo.c:3146]:
```suggestion
#if defined(OMRZTPF)

```

File: port/unix/omrsysinfo.c [hunk 6]
```diff
@@ -3252,144 +3239,105 @@ omrsysinfo_get_limit(struct OMRPortLibrary *portLibrary, uint32_t resourceID, ui
...
-#endif /* defined(AIXPPC) || (defined(LINUX) && !defined(OMRZTPF)) || defined(OSX) || defined(J9ZOS390) */
-	}
-	break;
-	default:
-		Trc_PRT_sysinfo_getLimit_unrecognised_resourceID(resourceID);
-		*limit = OMRPORT_LIMIT_UNKNOWN_VALUE;
-		rc = OMRPORT_LIMIT_UNKNOWN;
 	}
 
+
```
Comment (rwy0717) [inline port/unix/omrsysinfo.c:3244]:
```suggestion
```

File: port/unix/omrsysinfo.c [hunk 7]
```diff
@@ -3429,12 +3377,13 @@ getLimitSharedMemory(struct OMRPortLibrary *portLibrary, uint64_t *limit)
...
 	return OMRPORT_LIMIT_UNKNOWN;
-#else /* defined(LINUX) && !defined(OMRZTPF) */
+#else /* defined(LINUX) */
 	Trc_PRT_sysinfo_getLimitSharedMemory_notImplemented(OMRPORT_LIMIT_UNKNOWN);
 	*limit = OMRPORT_LIMIT_UNKNOWN_VALUE;
 	return OMRPORT_LIMIT_UNKNOWN;
-#endif /* defined(LINUX) && !defined(OMRZTPF) */
+#endif /* defined(LINUX) */
 }
+#endif /* defined(OMRZTPF) */
```
Comment (rwy0717) [inline port/unix/omrsysinfo.c:3386]:
```suggestion

#endif /* defined(OMRZTPF) */
```

File: port/unix/omrsysinfo.c [hunk 8]
```diff
@@ -3141,73 +3141,43 @@ omrsysinfo_get_hostname(struct OMRPortLibrary *portLibrary, char *buffer, size_t
 	return 0;
 }
 
+
```
Comment (rwy0717) [inline port/unix/omrsysinfo.c:3144]:
```suggestion
```

**Comment by rwy0717** [comment]
@genie-omr build all

**Pull request merged**